kotlin/src/main/kotlin/com/linecorp/armeria/common/kotlin/CoroutineContextAwareExecutor.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.kotlin
+
+import com.linecorp.armeria.common.ContextAwareExecutor
+import java.util.concurrent.Executor
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asCoroutineDispatcher
+
+/**
+ * Converts an instance of [ContextAwareExecutor] to an implementation of [CoroutineDispatcher].
+ * The returned [CoroutineContext] also contains an [ArmeriaRequestCoroutineContext] that automatically
+ * propagates the [ContextAwareExecutor.context] when the coroutine is resumed on a thread.
+ */
+fun ContextAwareExecutor.asCoroutineDispatcher(): CoroutineContext {
+    return (this as Executor).asCoroutineDispatcher() + context().asCoroutineContext()
```

Comment:
Should we unwrap `ContextAwareExecutor` and then combine the `CouroutineDispatcher` with `CoroutineContext` to avoid double pushes?
```suggestion
    return (this.withoutContext() as Executor).asCoroutineDispatcher() + context().asCoroutineContext()
```

Reply:
That's a good suggestion. 👍